DrinkManagerConsole/SearchDrinkByName.cs

```diff
@@ -0,0 +1,56 @@
+﻿using BLL;
+using System;
+using System.Collections.Generic;
+
+namespace DrinkManagerConsole
+{
+    internal class SearchDrinkByName
+    {
+        private readonly List<Drink> drinksListToSearch;
+
+        public SearchDrinkByName(List<Drink> drinksList)
+        {
+            drinksListToSearch = drinksList;
+        }
+
+        internal void SearchByName()
+        {
+            Console.Clear();
+            Console.WriteLine("\nSearch drink by name");
+            Console.WriteLine("-------------------------------");
+            Console.Write("Enter a drink name to find: ");
+
+            int numberOfDrinksFound = 0;
+            string textToSearch;
+            do
+            {
+                textToSearch = Console.ReadLine();
+            } while (textToSearch == null);
+
+            foreach (var drink in drinksListToSearch)
+            {
+                if (drink.Name.ToLower().Contains(textToSearch.ToLower()))
+                {
+                    Console.WriteLine("\n---------------------------------------------------------------------------------");
+                    Console.WriteLine($"Name: {drink.Name}");
+                    Console.WriteLine($"Is alcoholic: {drink.AlcoholicInfo}");
+                    Console.WriteLine($"Category: {drink.Category}");
+                    Console.WriteLine($"Glass type: {drink.GlassType}");
+                    Console.WriteLine($"Ingredients: ");
+                    foreach (var ingredient in drink.Ingredients)
+                    {
+                        if (ingredient.Name == null) continue;
+                        Console.WriteLine($"{ingredient.Name} : {ingredient.Amount}");
+                    }
+                    Console.WriteLine($"Instructions: {drink.Instructions}");
+                    numberOfDrinksFound++;
+                }
+            }
+
+            if (numberOfDrinksFound == 0)
+            {
+                Console.WriteLine("No matching drinks in our database.");
+            }
+        }
+    }
```

Comment:
I'll not go line by line here, but I'll give you a general idea what to fix -
First of all, you should separate the console from the logic of the app itself. I would suggest having a method that gets string that users put to search and then searching the drink and returning it (or null or whatever indicator that it wasn't found).

Also I think it would be good to put all drinks aggregation methods into one class-service. Maybe even call it `DrinkService` It will actually be useful in the part where you will learn about Dependency Injection ;)